File: iOS/Sources/Core/Extraction/Models/LineItem.swift
import Foundation

struct LineItem: Equatable, Codable {
    let name: String
    let quantity: Double
    let unitPricePaise: Int64
    let totalPricePaise: Int64
    let discountPaise: Int64

    init(name: String, quantity: Double, unitPricePaise: Int64, discountPaise: Int64 = 0) {
        self.name = name
        self.quantity = quantity
        self.unitPricePaise = unitPricePaise
        self.totalPricePaise = Int64(quantity * Double(unitPricePaise))
        self.discountPaise = discountPaise
    }
}

File: core/src/main/java/com/jinvoice/core/extraction/models/LineItem.kt
package com.jinvoice.core.extraction.models

/**
 * A single line item from an invoice.
 * All monetary values in paise (integer minor units) — no floats.
 */
data class LineItem(
    val name: String,
    val quantity: Double,
    val unitPricePaise: Long,
    val totalPricePaise: Long = (quantity * unitPricePaise).toLong(),
    val discountPaise: Long = 0L,
)
